bika/lims/browser/sample/view.py

```diff
@@ -484,25 +497,24 @@ def folderitem(self, obj, item, index):
         item['field']['getScheduledSamplingSampler'] =\
             'ScheduledSamplingSampler'
         # inline edits for Preserver and Date Preserved
-        checkPermission = self.context.portal_membership.checkPermission
-        if checkPermission(PreserveSample, obj):
-            item['required'] = ['getPreserver', 'getDatePreserved']
-            item['allow_edit'] = ['getPreserver', 'getDatePreserved']
-            preservers = getUsers(obj, ['Preserver', 'LabManager', 'Manager'])
-            getAuthenticatedMember = self.context.portal_membership.getAuthenticatedMember
-            username = getAuthenticatedMember().getUserName()
-            users = [({'ResultValue': u, 'ResultText': preservers.getValue(u)})
-                     for u in preservers]
-            item['choices'] = {'getPreserver': users}
-            preserver = username in preservers.keys() and username or ''
-            item['getPreserver'] = preserver
-            item['getDatePreserved'] = self.ulocalized_time(DateTime())
-            item['class']['getPreserver'] = 'provisional'
-            item['class']['getDatePreserved'] = 'provisional'
+        # if checkPermission(PreserveSample, obj):
```

Comment:
Note that the system was checking here for the permission `PreserveSample`, that is different from the permissions checked before in lines L466 and L472 (`SampleSample`), so the code below shouldn't be commented. Rather, I think is worth to spend some time trying to `group` all the logic between L454 and L513 in just one understandable chunk of code, cause there are a lot of similarities.

Reply:
@xispa I tried my best but found it harder than I expected at first. Here is the result, but I am not sure it is much better than the way it was before.

commit link: 54601a6